### PR TITLE
improve navigate bar (implement by component)

### PR DESCRIPTION
**Pull request by Kyl23**
https://github.com/Kyl23/to-do-list/issues/2

### CODE REVIEW

File: src/Action/ActionType.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+};
+const SWITCH_TASK = (id) => {
+  return { type: "SWITCH_TASK", id: id };
+};
+export const ActionType = {
+  INIT_TASK: INIT_TASK,
+  NEW_TASK: NEW_TASK,
+  DELETE_TASK: DELETE_TASK,
+  SWITCH_TASK: SWITCH_TASK,
+};
```
Comment (JiaAnTW) [inline src/Action/ActionType.js:28]: 建議再把type字串拉出來用變數管理，這樣reducer那邊就能共用

File: src/component/Reduxpage.js [hunk 2]
```diff
@@ -2,27 +2,34 @@ import { useEffect } from "react";
...
-            console.log(result.data.length);
-            dispatch({ type: "Value", value: result.data.length + 1 });
-            dispatch({ type: "InitTask", allTask: result.data });
-          });
+    const fetching = async () => {
+      if (tasks.id === 0) {
+        dispatch(getFromServer());
       }
     };
     fetching();
```
Comment (JiaAnTW) [inline src/component/Reduxpage.js]: 不需要再定義function了，直接呼叫定義內容就可以

File: src/component/contextAPI/header.js [hunk 3]
```diff
@@ -1,48 +1,49 @@
...
-        setTasks(pool);
-      }
+    //fetch one time only
+    const fetching = async () => {
+      const data = await getTasksFromServer();
+      SetOriginTasks(data);
     };
-    getTasks();
-  }, [id, pool, setTasks, isLegal, setId]);
+    fetching();
```
Comment (JiaAnTW) [inline src/component/contextAPI/header.js]: Same as https://github.com/Kyl23/to-do-list/pull/4/files#r756832955

File: src/component/contextAPI/input.js [hunk 4]
```diff
@@ -1,19 +1,32 @@
...
+  const n_id = tasks.id;
+  const SetTask = useCallback(
+    (value) => {
+      if (value.length <= 0) return;
+      let tp = {
+        id: n_id,
+        value: value,
+        reminder: false,
+      };
+      setTasks({ tasks: [...tasks.tasks, tp], id: n_id + 1 });
```
Comment (JiaAnTW) [inline src/component/contextAPI/input.js]: id的邏輯可以改成用useReducer封裝，就能在這裡避免還要思考n_id和task變動時會不會讓SetTask重新定義

File: src/index.js [hunk 5]
```diff
@@ -3,19 +3,20 @@ import ReactDOM from "react-dom";
...
-import { PoolData } from "./Pool/AppPool";
 import { Provider } from "react-redux";
 import allReducers from "../src/store";
-import { createStore } from "redux";
+import { createStore, applyMiddleware } from "redux";
+import { DataProvider } from "./context/HomePgContext";
+import thunk from "redux-thunk";
 
-const Store = createStore(allReducers);
+const Store = createStore(allReducers, applyMiddleware(thunk));
```
Comment (JiaAnTW) [inline src/index.js]: 非react component/class類別的命名用小寫，避免誤會

File: src/component/redux/put_to_do.js [hunk 6]
```diff
@@ -1,33 +1,19 @@
 import React from "react";
-import PropTypes from "prop-types";
 import { AddButton } from "../../style/css_element";
 import { useDispatch, useSelector } from "react-redux";
-function Put_To_Do() {
-  const text = useSelector((state) => state.input);
-  const id = useSelector((state) => state.id);
+import { ActionType } from "../../Action/ActionType";
+function Put_To_Do({ text, setText }) {
```
Comment (JiaAnTW) [inline src/component/redux/put_to_do.js]: Not a blocker, but just wonder why remove `PropTypes` here?

Comment (Kyl23) [reply src/component/redux/put_to_do.js]: when using redux it didn't any props need to define type , so remove it .But after first time review , props (text,setText) have been added back.And I forgot to use proptypes define the type.XD....

File: src/index.js (hunk 5 above)
Comment (Kyl23) [reply src/index.js]: Noted

File: src/component/Reduxpage.js (hunk 2 above)
Comment (Kyl23) [reply src/component/Reduxpage.js]: Noted

**Review by JiaAnTW — APPROVED** [review]

**Pull request merged**